src/supervisor-api.ts

```diff
@@ -208,17 +227,28 @@ export class SupervisorAPI {
 				log.debug('Supervisor API listening on allowed interfaces only');
 			}
 		} catch (err) {
-			log.error(
-				'Error on switching supervisor API listening rules - stopping API.\n',
-				err,
-			);
-			this.stop();
+			log.error('Error on switching supervisor API listening rules', err);
+			return this.stop({ errored: true });
 		}
 	}
 
-	public stop() {
+	public async stop(options?: SupervisorAPIStopOpts): Promise<void> {
 		if (this.server != null) {
-			this.server.close();
+			options?.errored
+				? log.error('Stopping Supervisor API...')
+				: log.debug('Stopping Supervisor API...');
```

Comment:
Honestly I don't feel that both a stopping and stopped message are needed, just a success and failure

Reply:
I added that log because the supervisor logs when it's starting the API so I thought it was appropriate to add when it's stopping in addition to started and stopped message if successful.

https://github.com/balena-io/balena-supervisor/blob/ce39921c7ba6bb80f63e5f51750b432b7c6a17ee/src/supervisor.ts#L120-L125

Reply:
I removed these logs because I think it's more appropriate for the function calling the stop to log that it is stopping the supervisor api like in the above example when it starts the supervisor api.